cpp/src/arrow/compute/kernels/scalar_set_lookup_test.cc

```diff
@@ -253,11 +268,11 @@ TEST_F(TestIsInKernel, TimeDuration) {
               "[true, false, false, true, true]", /*skip_nulls=*/true);
   }
 
-  // Different units, invalid cast
-  ASSERT_RAISES(Invalid, IsIn(ArrayFromJSON(duration(TimeUnit::SECOND), "[0, 1, 2]"),
-                              ArrayFromJSON(duration(TimeUnit::MILLI), "[0, 2]")));
+  // Different units, cast value_set to values
```

Comment:
Is the comment correct? It will attempt to cast value_set to values, fail because of truncation, and then cast values to value_set.

Reply:
Comment updated.